### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by MySmilingTurban**
Fixes issues with sign in with google and saving state with end session button.

### CODE REVIEW

File: src/auth/index.tsx [hunk 1]
```diff
@@ -189,6 +221,7 @@ export const AuthContextProvider = ({ children }: { children: ReactElement }) =>
       if (currentUser !== null) {
         const { uid, emailVerified, metadata } = currentUser as FirebaseUser;
         const userDetails = await getUserData(uid);
+        console.log('UserDetails2inHook', userDetails);
```
Comment (amitojsingh) [inline src/auth/index.tsx:224]: can we remove this console?

File: src/components/buttons/EndSessionBtn.tsx [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
+import { ProgressData } from 'types';
 
-const EndSessionButton = ({ className = '' }: { className: string }) => {
+const EndSessionButton = ({ uid, currentData, className = '' }: { uid: string, currentData: ProgressData, className: string }) => {
   const navigate = useNavigate();
+  const [saving, toggleSaving] = useState(false);
+  
+  const endSession = async (route: string) => {
+    toggleSaving(true);
+    console.log('Current Data: ', currentData);
```
Comment (amitojsingh) [inline src/components/buttons/EndSessionBtn.tsx:14]: console

File: src/database/shabadavalidb/words.ts [hunk 3]
```diff
@@ -56,14 +56,19 @@ export const getWords = async (uid: string, isLearnt: boolean) => {
       limit(CONSTANTS.GET_WORD_LIMIT),
     );
     const querySnapshot = await getDocs(q);
+    console.log('querySnapshot:', querySnapshot);
```
Comment (amitojsingh) [inline src/database/shabadavalidb/words.ts:59]: console

Comment (amitojsingh) [reply src/database/shabadavalidb/words.ts:59]: in the whole page.

**Review by amitojsingh — CHANGES_REQUESTED** [review]

File: src/auth/index.tsx [hunk 4]
```diff
@@ -60,46 +68,69 @@ export const AuthContextProvider = ({ children }: { children: ReactElement }) =>
...
-        lastLogInAt: Timestamp.now(),
-      } as User;
-      setUser(userData);
-      setLoading(false);
-      return true;
+          lastLogInAt: Timestamp.now(),
+        } as User;
+        console.log('sign in with google', userData);
+        if (uid) await setWordIds(uid);
+        dispatch(setCurrentGamePosition(userData.progress.currentProgress));
```
Comment (amitojsingh) [inline src/auth/index.tsx:120]: don't thing veer ji we should add this in useEffect.

Comment (MySmilingTurban) [reply src/auth/index.tsx:120]: I just tested it after removing the part from useEffect, it doesn't work.
But when I removed it from signInWithGoogle function but not from useEffect it works.

File: src/auth/index.tsx [hunk 5]
```diff
@@ -60,46 +68,69 @@ export const AuthContextProvider = ({ children }: { children: ReactElement }) =>
...
+            coins: 0,
+            progress: {
+              currentProgress: 0,
+              gameSession: [],
+              currentLevel: 0,
+            },
+            created_at: Timestamp.now(),
+            updated_at: Timestamp.now(),
+            lastLogInAt: Timestamp.now(),
+            user: null as FirebaseUser | null,
```
Comment (gurjit03) [inline src/auth/index.tsx]: @MySmilingTurban Any chance we can avoid this `typecasting` as well? 
https://medium.com/livefront/to-cast-or-not-to-cast-a-typescript-dilemma-c0c20c53c6d9

File: src/auth/index.tsx (hunk 1 above)
Comment (gurjit03) [inline src/auth/index.tsx]: @MySmilingTurban stale log here !

File: src/auth/index.tsx [hunk 6]
```diff
@@ -60,46 +68,69 @@ export const AuthContextProvider = ({ children }: { children: ReactElement }) =>
...
-      return true;
+          lastLogInAt: Timestamp.now(),
+        } as User;
+        console.log('sign in with google', userData);
+        if (uid) await setWordIds(uid);
+        dispatch(setCurrentGamePosition(userData.progress.currentProgress));
+        dispatch(setCurrentLevel(userData.progress.currentLevel));
+        dispatch(setNanakCoin(userData.coins));
+        dispatch(addScreens(userData.progress.gameSession));
+        dispatch(addNextScreens(userData.nextSession ?? []));
```
Comment (gurjit03) [inline src/auth/index.tsx]: @MySmilingTurban Can see too many dispatch calls and also similar calls below, just was wondering can we combine those ?

**Pull request closed without merging**